### ISSUE TITLE
CreateLayer change the CS

### ISSUE THREAD

**Issue by aharondavid** [OSGeo/gdal#1857]
## Expected behavior and actual behavior.

char* WKT = "PROJCS["MSK_Cherepovec",GEOGCS["Pulkovo 1942",DATUM["Pulkovo_1942",SPHEROID["Krassowsky 1940",6378245,298.3,AUTHORITY["EPSG","7024"]],TOWGS84[23.92,-141.27,-80.9,0,0.35,0.82,-0.12],AUTHORITY["EPSG","6284"]],PRIMEM["Greenwich",0,AUTHORITY["EPSG","8901"]],UNIT["degree",0.0174532925199433,AUTHORITY["EPSG","9122"]],AUTHORITY["EPSG","4284"]],PROJECTION["Transverse_Mercator"],PARAMETER["latitude_of_origin",0],PARAMETER["central_meridian",36.81282638],PARAMETER["scale_factor",1.000003232221],PARAMETER["false_easting",-48241.5913],PARAMETER["false_northing",-6550669.853900001],UNIT["metre",1,AUTHORITY["EPSG","9001"]],AUTHORITY["EPSG","2552"]]"

        OGRSpatialReference SpatialReference;                
                SpatialReference.importFromWkt(&pWKT);
        pSpatialReference = SpatialReference.Clone();

/* m_pDataSourceCreated is GPKG new Data Source */
m_pDataSourceCreated->CreateLayer(ST2A_CP(sLayerName, CP_UTF8), pSpatialReference, layerGeomType, papszOptions);

The created GPKG data source contain a different  CS :

PROJCS["Pulkovo 1942 / 3-degree Gauss-Kruger zone 35",
    GEOGCS["Pulkovo 1942",
        DATUM["Pulkovo_1942",
            SPHEROID["Krassowsky 1940",6378245,298.3,
                AUTHORITY["EPSG","7024"]],
            TOWGS84[23.92,-141.27,-80.9,0,0.35,0.82,-0.12],
            AUTHORITY["EPSG","6284"]],
        PRIMEM["Greenwich",0,
            AUTHORITY["EPSG","8901"]],
        UNIT["degree",0.0174532925199433,
            AUTHORITY["EPSG","9122"]],
        AUTHORITY["EPSG","4284"]],
    PROJECTION["Transverse_Mercator"],
    PARAMETER["latitude_of_origin",0],
    PARAMETER["central_meridian",105],
    PARAMETER["scale_factor",1],
    PARAMETER["false_easting",35500000],
    PARAMETER["false_northing",0],
    UNIT["metre",1,
        AUTHORITY["EPSG","9001"]],
    AUTHORITY["EPSG","2552"]]

## Operating system

Windows 10

## GDAL version and provenance

GDAL 2.2.2 
GDAL 3.1

**Comment by rouault** [comment]
Yes, this is expected behaviour. Your input WKT contains a top-level AUTHORITY["EPSG","2552"] node, so the driver assumes the WKT is consistent with the official representation of EPSG:2552 and will insert a standard code in the gpkg_spatial_ref_sys table. If you need your original definition to be preserved, you need to remove the AUTHORITY["EPSG","2552"]  node

**Comment by rouault** [comment]
Actually I should have mentioned that the CRS definition (if one looks at the projection parameters) in your WKT is highly different from the one of EPSG:2552. This is a very dangerous situation.

**Comment by aharondavid** [comment]
there is an in consist way in who it is works:
in one hand, when I read a shape file CS and convert its coordinates with the above WKT, the conversion result is according to customize WKT above. 
In the other hand, when I create a new layer with the same CS, I get a new layer with the new CS.
when I convert the same coordinates from this new layer, I get a different result

**Comment by aharondavid** [comment]
I think, what it is needed is a flag to CreateLayer that say - do not change the given customized CS

**Comment by jratike80** [comment]
You have a WKT that states that it is EPSG:2552 and on the other hand it contains different parameters that define a custom coordinate system. I doubt if such CS worth is keeping. But should GDAL throw an error about inconsistent CS?

**Comment by aharondavid** [comment]
GDAL 2.2.2 and GDAL 3 doesn't throw any exception for this customize WKT and the behave is as I mentioned above

**Comment by rouault** [comment]
> But should GDAL throw an error about inconsistent CS?

It would be nice to have, but there's a risk to too sensitive. There might be some tiny differences between equivalent representations (names, some differences on last decimals) . OGRSpatialReference::IsSame() has some provision for such fuzzy comparison though. A warning would be less risky than an error.

**Comment by aharondavid** [comment]
By the way, OGRSpatialReference::exportToWkt return the above customized WKT as it is, so
why not to use this returned WKT for injection it into the new created GPKG

**Comment by aharondavid** [comment]
another thing, I just check it on create SQlite layer instead of GPKG, and it works.

**Comment by jratike80** [comment]
Does that invalid WKT come from some popular software? I mean, do you think that there are lots of users affected? If you know the software it would be kind to ask them to fix their bug.

**Comment by aharondavid** [comment]
The invalid WKT is not the issue, the issue is:
1. User customize a (invalid) WKT in shapefile. GDAL and PROJ doesn't inform about any error
2. then, he convert the shapefile coordinates and get some result
3. then, the user use in the same WKT to create GPKG, but, now, converting the same coordinates
yield a different result

**Comment by aharondavid** [comment]
Hello @rouault , I just look briefly into your change, should I use  in CreateLayer Options flag - OGR_GPKG_CHECK_SRS=YES  ?

**Comment by rouault** [comment]
> should I use in CreateLayer Options flag - OGR_GPKG_CHECK_SRS=YES ?

no, this is the default value (this is a configuration option / environment variable, not a creation option), and this default checks if the passed SRS is consistent with the official code it might refer to, and if not, it will createa new entry in the database for it. This config option is just a provision if this new behaviour would cause issue
